Makefile

```diff
@@ -114,6 +114,79 @@ export ROCM_LDFLAGS := -L$(ROCM_LIBDIR) -lamdhip64
+AOT_HIP_FLAGS     := -fsycl-targets=amdgcn-amd-amdhsa -Xsycl-target-backend --offload-arch=gfx900 --rocm-path=$(ROCM_BASE) -Wno-linker-warnings 
+
+# INTEL flags: compile AOT for all the CPUs and for the GPU on olice-05
+# CUDA flags : compile AOT for for NVIDIA GPUs
+# HIP flags  : compile AOT for architectures with ID gfx900 (e.g. the Radeon PRO WX 9100)
+
+# -Wno-linker-warnings will not be needed be needed anymore with https://github.com/intel/llvm/pull/7245
+
+ifdef USE_SYCL_ONEAPI
+ONEAPI_BASE       := /cvmfs/projects.cern.ch/intelsw/oneAPI/linux/x86_64/2022
+ONEAPI_VERSION    := latest
+TBB_BASE          := $(ONEAPI_BASE)/tbb/latest
+TBB_LIBDIR        := $(TBB_BASE)/lib/intel64/gcc4.8
+ifneq ($(wildcard $(ONEAPI_BASE)),)
+ONEAPI_ENV        := $(ONEAPI_BASE)/setvars.sh # --config="/eos/user/a/aperego/dev/pixeltrack-standalone/config.txt"
+                                               # the config.txt file can be used to source only specific tools 
+                                               # or a specific version of a tool of the oneAPI package
+SYCL_BASE         := $(ONEAPI_BASE)/compiler/$(ONEAPI_VERSION)/linux
+USER_SYCLFLAGS    := -fp-model=precise -fimf-arch-consistency=true -no-fma
+# math flags : -fp-model=precise -fimf-arch-consistency=true -no-fma
+# workaround for the unexpected intrinsic in ONEAPI 2022.2.0 (SYCL BUG): -fno-sycl-early-optimizations
+export SYCL_CXX      := $(SYCL_BASE)/bin/dpcpp
+export SYCL_CXXFLAGS := -O3 -fsycl -Wno-sycl-strict $(filter-out $(LLVM_UNSUPPORTED_CXXFLAGS),$(CXXFLAGS)) $(USER_SYCLFLAGS)
+endif
+
+else
+# use llvm 
+SYCL_BASE      := /cvmfs/patatrack.cern.ch/externals/x86_64/rhel8/intel/sycl/build-2022-09
+USER_SYCLFLAGS := 
+
+# make CPUs visible
+export OCL_ICD_FILENAMES := /cvmfs/patatrack.cern.ch/externals/x86_64/rhel8/intel/sycl/runtime/intel/oclcpuexp_2022.14.8.0.04/x64/libintelocl.so
+
+export SYCL_CXX      := $(SYCL_BASE)/bin/clang++
+export SYCL_CXXFLAGS := -O3 -fsycl $(filter-out $(LLVM_UNSUPPORTED_CXXFLAGS),$(CXXFLAGS)) $(USER_SYCLFLAGS)
+endif
+
+# Now add the flags to compile ahead of time for CPUs, Intel GPUs, NVIDIA GPUs and AMD GPUs
+# The flags for NVIDIA GPUs and AMD GPUs are added only if llvm is used since they are not yet supported by dpcpp
+# At the moment it's not possible to compile AOT for both CUDA and AMD together (LLVM BUG)
+# so if both are there the default is to compile only for the CUDA backend
+# (the AMD backend has some bugs so there is a high probability that it won't even compile)
+
+ifdef ONEAPI_BASE
+SYCL_CXXFLAGS += $(AOT_INTEL_FLAGS)
+else
+ifdef CUDA_BASE
+SYCL_CXXFLAGS += $(AOT_CUDA_FLAGS)
+else 
+ifdef ROCM_BASE
+SYCL_CXXFLAGS += $(AOT_HIP_FLAGS)
+endif
+endif
+endif
+# check if libraries are under lib or lib64
+ifdef SYCL_BASE
+ifneq ($(wildcard $(SYCL_BASE)/lib/libsycl.so),)
+SYCL_LIBDIR := $(SYCL_BASE)/lib
+else ifneq ($(wildcard $(SYCL_BASE)/lib64/libsycl.so),)
+SYCL_LIBDIR := $(SYCL_BASE)/lib64
+else
+SYCL_BASE :=
```

Comment:
Undefining `SYCL_BASE` basically means disabling SYCL. Should the `TBB_BASE` be explicitly undefined as well in this case?